### PR TITLE
Verification instructions

### LINKED ISSUE #259: Add guide on verifying integrity of downloaded files
It would be helpful for users to have explicit directions as to how to verify downloaded files by using the sha256 checksums listed alongside each download.

I don't know if we want this as a specific guide under installation (or whereever), or as a section on the downloads page, but it's something which is often included in addition to supplying the checksums of files (see "ledger-live" as an example: https://www.ledger.com/ledger-live/lld-signatures)

To produce a checksum of the downloaded file, the user can run:

Linux:
```bash
sha256sum <path-to-downloaded-file>
```

MacOS: 
```bash
shasum -a 256 <path-to-downloaded-file>
```

Windows (powershell): 
```powershell
Get-FileHash <path-to-downloaded-file> -Algorithm SHA256
```

Furthermore, desktop wallet binaries are signed in order to be able to verify that the binary is indeed an official release from Concordium. Steps on how to verify against the produced signature should also be included here.

To verify a downloaded file (\<file-path\> placeholder in code block) using the provided signature, the user will need:
- The signature file (\<signature-file-path\> placeholder in code block)
- The public key matching the private key used to produce the signature (\<pubkey-file-path\> placeholder in code block)

The file can then be verified by running the following in a terminal:
```bash
openssl dgst -sha256 -verify <pubkey-file-path> -signature <signature-file-path> <file-path>
```

**Sub-tasks**

- [ ] Verify by comparing with published checksum (all downloads with published checksum)
- [ ] Verify with published signature (DTW)

### PR DESCRIPTION
**Pull request by soerenbf**
## Purpose

The purpose of the proposed changes is, to aid the user in verifying the integrity of downloaded files. For many of our releases, we publish a sha256 checksum. This describes the steps required for a user to execute, to verify that the downloaded file matches what is released by Concordium.

Specifically for the desktop wallet, we also publish a signature of each release. A description of how to verify that the downloaded desktop wallet release has been signed by Concordium is included for each OS/target.

Variables have been created for file names, hashes, etc. in order to easily update documentation to match new releases.

**Notes**
- Public key for desktop wallet doesn't exist at the url defined in`extlinks` in `conf.py`. This has to be published the first time we sign the desktop wallet.
- Signatures do not exist in github releases yet, but is a part of releasing updates for the desktop wallet past **v1.2**

## Changes

- Added verification page (next to downloads page) with verification steps for all releases that have checksum published
- Added links to respective verification sections alongside checksum of each release
- Added variables (`variables.rst`, mainnet only) for parameters related to verification.
- Added external links used in verification instructions in `conf.py` in variable `extlinks` (used by sphinx.ext.extlinks extension) 

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.

Closes #259

### CODE REVIEW

File: source/mainnet/net/installation/cdw-verification-instructions/cdw-appimage.rst [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+.. code-block:: console
+    :substitutions:
+
+    $sha256sum |cdw-appimage|
+    |cdw-appimage-checksum|
+
+**Verify download with signature**
+
+To verify that the downloaded file is indeed officially released by concordium, you can verify that the file is signed by Concordium. To do this, you need a signature of the release and a public key.
```
Comment (jens-concordium) [inline source/mainnet/net/installation/cdw-verification-instructions/cdw-appimage.rst:19]:
```suggestion
To verify that the downloaded file is an official Concordium release, you can verify that the file is signed by Concordium. To do this, you need a signature of the release and a public key.
```

File: source/mainnet/net/installation/cdw-verification-instructions/cdw-deb.rst [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+.. code-block:: console
+    :substitutions:
+
+    $sha256sum |cdw-deb|
+    |cdw-deb-checksum|
+
+**Verify download with signature**
+
+To verify that the downloaded file is indeed officially released by concordium, you can verify that the file is signed by Concordium. To do this, you need a signature of the release and a public key.
```
Comment (jens-concordium) [inline source/mainnet/net/installation/cdw-verification-instructions/cdw-deb.rst:19]:
```suggestion
To verify that the downloaded file is an official Concordium release, you can verify that the file is signed by Concordium. To do this, you need a signature of the release and a public key.
```

File: source/mainnet/net/installation/cdw-verification-instructions/cdw-macos.rst [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+.. code-block:: console
+    :substitutions:
+
+    $shasum -a 256 |cdw-mac|
+    |cdw-mac-checksum|
+
+**Verify download with signature**
+
+To verify that the downloaded file is indeed officially released by concordium, you can verify that the file is signed by Concordium. To do this, you need a signature of the release and a public key.
```
Comment (jens-concordium) [inline source/mainnet/net/installation/cdw-verification-instructions/cdw-macos.rst:17]:
```suggestion
To verify that the downloaded file is an official Concordium release, you can verify that the file is signed by Concordium. To do this, you need a signature of the release and a public key.
```

File: source/mainnet/net/installation/cdw-verification-instructions/cdw-rpm.rst [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+.. code-block:: console
+    :substitutions:
+
+    $sha256sum |cdw-rpm|
+    |cdw-rpm-checksum|
+
+**Verify download with signature**
+
+To verify that the downloaded file is indeed officially released by concordium, you can verify that the file is signed by Concordium. To do this, you need a signature of the release and a public key.
```
Comment (jens-concordium) [inline source/mainnet/net/installation/cdw-verification-instructions/cdw-rpm.rst:19]:
```suggestion
To verify that the downloaded file is an official Concordium release, you can verify that the file is signed by Concordium. To do this, you need a signature of the release and a public key.
```

File: source/mainnet/net/installation/cdw-verification-instructions/cdw-windows.rst [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+.. code-block:: console
+    :substitutions:
+
+    $Get-FileHash |cdw-windows| -Algorithm SHA256
+    |cdw-windows-checksum|
+
+**Verify download with signature**
+
+To verify that the downloaded file is indeed officially released by concordium, you can verify that the file is signed by Concordium. To do this, you need a signature of the release and a public key.
```
Comment (jens-concordium) [inline source/mainnet/net/installation/cdw-verification-instructions/cdw-windows.rst:18]:
```suggestion
To verify that the downloaded file is an official Concordium release, you can verify that the file is signed by Concordium. To do this, you need a signature of the release and a public key.
```

**Review by jens-concordium — COMMENTED** [review]
I just had one suggestion for the wording in one of the sentences. I didn't try to verify the signature and public key for the DW, as they weren't available via the link.

**Comment by jens-concordium** [comment]
On further thought, let's wait with this, until after 1.1.3 is released.

**Comment by soerenbf** [comment]
This has been updated with whichever versions were in the main branch.

File: source/mainnet/net/installation/cdw-verification-instructions/cdw-appimage.rst [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    |cdw-appimage-checksum|
+
+**Verify download with signature**
+
+To verify that the downloaded file is an official Concordium release, you can verify that the file is signed by Concordium. To do this, you need a signature of the release and a public key.
+
+* :cdw-sig:`Download signature <AppImage>`
+* :cdw-pubkey:`Download public key <>`
+
+With the **signature**, the **public key**, and the **downloaded file** all in the same directory, executing the following steps will verify that the file has been signed by Concordium.
```
Comment (dg-concordium) [inline source/mainnet/net/installation/cdw-verification-instructions/cdw-appimage.rst:24]: With the **signature**, the **public key**, and the **downloaded file** all in the same directory, execute the following steps to verify that the file has been signed by Concordium.

File: source/mainnet/net/installation/cdw-verification-instructions/cdw-appimage.rst [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+In a terminal:
+
+#. Navigate to the directory containing the assets needed to verify.
+#. Then paste the first line of the following block into the terminal
+#. The command should output **Verified OK** as a result, as inidicated by the second line in the block.
```
Comment (dg-concordium) [inline source/mainnet/net/installation/cdw-verification-instructions/cdw-appimage.rst:30]: The command outputs **Verified OK** as a result, as indicated by the second line in the block.

File: source/mainnet/net/installation/cdw-verification-instructions/cdw-deb.rst [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    |cdw-deb-checksum|
+
+**Verify download with signature**
+
+To verify that the downloaded file is an official Concordium release, you can verify that the file is signed by Concordium. To do this, you need a signature of the release and a public key.
+
+* :cdw-sig:`Download signature <deb>`
+* :cdw-pubkey:`Download public key <>`
+
+With the **signature**, the **public key**, and the **downloaded file** all in the same directory, executing the following steps will verify that the file has been signed by Concordium.
```
Comment (dg-concordium) [inline source/mainnet/net/installation/cdw-verification-instructions/cdw-deb.rst:24]: With the **signature**, the **public key**, and the **downloaded file** all in the same directory, execute the following steps to verify that the file has been signed by Concordium.

File: source/mainnet/net/installation/cdw-verification-instructions/cdw-deb.rst [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
+In a terminal:
+
+#. Navigate to the directory containing the assets needed to verify.
+#. Then paste the first line of the following block into the terminal
+#. The command should output **Verified OK** as a result, as inidicated by the second line in the block.
```
Comment (dg-concordium) [inline source/mainnet/net/installation/cdw-verification-instructions/cdw-deb.rst:30]: The command outputs **Verified OK** as a result, as indicated by the second line in the block.

File: source/mainnet/net/installation/cdw-verification-instructions/cdw-rpm.rst [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+    |cdw-rpm-checksum|
+
+**Verify download with signature**
+
+To verify that the downloaded file is an official Concordium release, you can verify that the file is signed by Concordium. To do this, you need a signature of the release and a public key.
+
+* :cdw-sig:`Download signature <rpm>`
+* :cdw-pubkey:`Download public key <>`
+
+With the **signature**, the **public key**, and the **downloaded file** all in the same directory, executing the following steps will verify that the file has been signed by Concordium.
```
Comment (dg-concordium) [inline source/mainnet/net/installation/cdw-verification-instructions/cdw-rpm.rst:24]: With the **signature**, the **public key**, and the **downloaded file** all in the same directory, execute the following steps to verify that the file has been signed by Concordium.

File: source/mainnet/net/installation/cdw-verification-instructions/cdw-rpm.rst [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+
+In a terminal:
+
+#. Navigate to the directory containing the assets needed to verify.
+#. Then paste the first line of the following block into the terminal
+#. The command should output **Verified OK** as a result, as inidicated by the second line in the block.
```
Comment (dg-concordium) [inline source/mainnet/net/installation/cdw-verification-instructions/cdw-rpm.rst:30]: The command outputs **Verified OK** as a result, as indicated by the second line in the block.

File: source/mainnet/net/installation/verification-instructions.rst [hunk 12]
```diff
@@ -0,0 +1,111 @@
...
+=========================
+Verification instructions
+=========================
+
+.. contents::
+   :local:
+   :backlinks: none
+   :depth: 2
+
+To find instructions on how to verify the integrity of the file you have downloaded, navigate to the respective section, which will outline the steps needed for that specific file.
```
Comment (dg-concordium) [inline source/mainnet/net/installation/verification-instructions.rst:13]: To find instructions on how to verify the integrity of the file you have downloaded, navigate to the respective section that outlines the steps needed for that specific file.

**Review by dg-concordium — COMMENTED** [review]
I have made some small comments, just grammar, spelling, and general readability. You probably want to make these small changes before merging.

**Review by dg-concordium — APPROVED** [review]

**Pull request merged**